src/test/munit/oracle/arrays-and-struct-source-oracle-test-case.xml

```diff
@@ -0,0 +1,167 @@
+
+            <munit-tools:dequeue queueName="sourceLocalTransactionTestQueue" timeout="100" timeoutUnit="SECONDS"/>
+
+        </munit:execution>
+
+        <munit:validation>
+            <munit-tools:assert-equals actual="#[payload.list[0][1]]" expected="#[1]"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="prepareArrayAndStructLocalTransactionSourceWithExternalConnection" ignore="#[java!org::mule::extension::db::DbMunitUtils::isTestIgnored('oracle')]">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="dbListenerFlowExternalConnection"/>
+        </munit:enable-flow-sources>
+
+        <munit:execution>
+            <db:insert config-ref="${config}">
+                <db:sql><![CDATA[INSERT INTO TEST_LISTENER(ID) VALUES(1)]]></db:sql>
+            </db:insert>
+
+            <munit-tools:dequeue queueName="sourceLocalTransactionTestQueueExternalConnection" timeout="100" timeoutUnit="SECONDS"/>
+
+        </munit:execution>
+
+        <munit:validation>
+            <munit-tools:assert-equals actual="#[payload.list[0][1]]" expected="#[1]"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:after-suite name="dropStuff">
+        <try>
+            <db:execute-script config-ref="${config}">
+                <db:sql>
+                    drop procedure retain_first_struct;
+                    drop type TEST_STRUCT_COLLECTION;
+                    drop type TEST_STRUCT;
+                </db:sql>
+            </db:execute-script>
+            <error-handler>
+                <on-error-continue logException="false"/>
+            </error-handler>
+        </try>
+    </munit:after-suite>
+
+    <flow name="dbListenerFlow">
+        <db:listener config-ref="${config}" table="TEST_LISTENER" transactionalAction="ALWAYS_BEGIN">
+            <scheduling-strategy>
+                <fixed-frequency  frequency="10"/>
+            </scheduling-strategy>
+        </db:listener>
+
+        <set-variable
+                value="#[Db::prepareArray('TEST_STRUCT_COLLECTION', [ Db::prepareStruct('TEST_STRUCT', ['hi', 1, 'bye']), Db::prepareStruct('TEST_STRUCT', ['hi2', 2, 'bye2']) ])]"
+                variableName="list"/>
+
+        <db:stored-procedure config-ref="${config}" inOutParameters='#[{ "list": vars.list }]'>
+            <db:sql>call retain_first_struct(:list)</db:sql>
+        </db:stored-procedure>
+
+        <set-payload value="#[payload]"/>
+        <logger level="INFO"  message="#[payload]"/>
+
+        <munit-tools:queue queueName="sourceLocalTransactionTestQueue"/>
+    </flow>
+
+    <flow name="dbListenerFlowExternalConnection">
+        <db:listener config-ref="${config}" table="TEST_LISTENER" transactionalAction="ALWAYS_BEGIN">
+            <scheduling-strategy>
+                <fixed-frequency  frequency="10"/>
+            </scheduling-strategy>
+        </db:listener>
+
+        <db:insert config-ref="oracleDbPoolingListenerExternalConfig">
```

Comment:
I think is accurate that the external operation also uses the prepareArray. Please, move it after the set variable.